plugins/catalog/src/components/CatalogPage/CatalogPage.tsx

```diff
@@ -71,16 +97,22 @@ export const CatalogPage: FC<{}> = () => {
   );
 
   const { value, error, loading } = useAsync(async () => {
-    const filter = entityFilters[selectedFilter.id];
-    const all = await catalogApi.getEntities();
-    return all.filter(e => filter(e, { isStarred: isStarredEntity(e) }));
-  }, [selectedFilter.id, starredEntities.size]);
+    return await catalogApi.getEntities();
+  });
 
   const onFilterSelected = useCallback(
     selected => setSelectedFilter(selected),
     [],
   );
 
+  const filteredEntities = useMemo(() => {
+    const typeFilter = entityFilters[EntityFilterType.TYPE];
+    const leftMenuFilter = entityFilters[selectedFilter.id];
+    return value
+      ?.filter(e => leftMenuFilter(e, { isStarred: isStarredEntity(e) }))
+      .filter(e => typeFilter(e, { type: selectedTab }));
+  }, [selectedFilter.id, selectedTab, isStarredEntity, value?.filter]);
```

Comment:
`value.filter` feels wrong here conceptually, maybe just `value`? Like in "if value changes, recalculate this"

Reply:
> `value.filter` feels wrong here conceptually, maybe just `value`? Like in "if value changes, recalculate this"

Yea i agree. The linter hated that though. This exhaustive-deps rule is weird and i don't understand it fully. Leaving it out suggested me adding what's there now, and adding just value as you suggested gave this error `React Hook useMemo has an unnecessary dependency: 'value'. Either exclude it or remove the dependency array`